server/fishtest/static/js/application.js

```diff
@@ -225,9 +225,13 @@ function raise_for_status(response) {
 }
 
 async function fetch_json(url, options) {
-  const response = await fetch(url, options);
-  raise_for_status(response);
-  return response.json();
+  try {
+    const response = await fetch(url, options);
+    raise_for_status(response);
+    return response.json();
+  } catch (error) {
+    console.error("Network error: ", error);
+  }
```

Comment:
Is there a compelling reason to change the error behavior of the wrapper functions (returning null instead of an exception)?

Also, this changes the error handling in the caller. Did you check this has no impact on the current use of the wrapper functions (e.g. in notifications.js)?